Cabal/Distribution/Simple/Program/GHC.hs

```diff
@@ -83,6 +84,12 @@ data GhcOptions = GhcOptions {
   -- (we need to handle backwards compatibility.)
   ghcOptThisUnitId   :: Flag String,
 
+  -- | GHC doesn't make any assumptions about the format of
+  -- definite unit ids, so when we are instantiating a package it
+  -- needs to be told explicitly what the component being instantiated
+  -- is.  This only gets set when 'ghcOptInstantiatedWith' is non-empty
+  ghcOptThisComponentId :: Flag ComponentId,
+
   ghcOptInstantiatedWith :: [(ModuleName, OpenModule)],
```

Comment:
This is a good opportunity to add a Haddock comment to `ghcOptInstantiatedWith` as well.